piccolo/query/mixins.py

```diff
@@ -613,12 +616,132 @@ def __str__(self):
 @dataclass
 class GroupByDelegate:
     """
-    Used to group results - needed when doing aggregation.
+    Used to group results - needed when doing aggregation::
+
+        .group_by(Band.name)
 
-    .group_by(Band.name)
     """
 
     _group_by: t.Optional[GroupBy] = None
 
     def group_by(self, *columns: Column):
         self._group_by = GroupBy(columns=columns)
+
+
+class OnConflictAction(str, Enum):
+    do_nothing = "DO NOTHING"
+    do_update = "DO UPDATE"
+
+
+@dataclass
+class OnConflict:
+    target: t.Optional[t.Sequence[t.Union[str, Column]]] = None
+    action: t.Optional[OnConflictAction] = None
+    values: t.Optional[
+        t.List[t.Union[Column, t.Tuple[t.Union[str, Column], t.Any]]]
+    ] = None
+
+    @property
+    def target_string(self) -> str:
+        assert self.target
+
+        def to_string(value) -> str:
+            if isinstance(value, Column):
+                return f'"{value._meta.name}"'
+            elif isinstance(value, str):
+                return value
+            else:
+                raise ValueError("OnConflict.target isn't a valid type")
+
+        columns_str = ", ".join([to_string(i) for i in self.target])
+        return f"({columns_str})"
+
+    @property
+    def action_string(self) -> QueryString:
+        action = self.action
+        if isinstance(action, OnConflictAction):
+            if action == OnConflictAction.do_nothing:
+                return QueryString(OnConflictAction.do_nothing.value)
+            elif action == OnConflictAction.do_update:
+                values = []
+                query = f"{OnConflictAction.do_update.value} SET"
+
+                if not self.values:
+                    raise ValueError("No values specified for `on conflict`")
+
+                for value in self.values:
+                    if isinstance(value, Column):
+                        column_name = value._meta.db_column_name
+                        query += f' "{column_name}"=EXCLUDED."{column_name}"'
```

Comment:
```suggestion
                        query += f' "{column_name}"=EXCLUDED."{column_name}",'
```
for multiple columns